### PR TITLE
Format dates in a single place, Display build date instead of full commit hash in version info

### LINKED ISSUE #1933: Display build date instead of full commit hash in version info
1. Remove VITE_TAG
1. Remove VITE_COMMIT
1. fetch info from Spring API `/actuator/info` `{commit, tag, buildDate: timestamp}`
1. If tag is a version (v0.5.0) fetch latest version from github and compare with tag.
6. If tag is a commit we need to display formatted timestamp (`/api/info/timestampformat/iso`)
7. if tag contains `-SNAPSHOT` - display formatted timestamp
9. Keep the short one with the link

### LINKED ISSUE #2163: Make message timestamp format configurable
We changed default date format as requested in #2162.

We need the datetime format configurable on the frontend for the ones who prefer the US format or anything else.

### LINKED ISSUE #2566: Format dates in a single place
### Describe the solution you'd like

1. Using `/api/info/timestampformat` endpoint fetch info about system preferable datetime format.
2. Create utility function(helper) for date formatting.
3. Replace current formatted dates in app with new approach.

### PR DESCRIPTION
**Pull request by habrahamyanpro**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
- fixes #2566 
- fixes #1933
- fixes #2163

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:82]: I see no need of getting timeStampFormat in every place we use `formatTimestamp`. 

Please load `timeStampFormat` once and use inside of `formatTimestamp` helper

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:8]:
```suggestion
import { TimeStampFormat } from 'generated-sources';
```

**Review by workshur — CHANGES_REQUESTED** [review]
Please follow my review comments and fix linting issues

File: kafka-ui-react-app/src/lib/constants.ts
Comment (habrahamyanpro) [reply kafka-ui-react-app/src/lib/constants.ts:59]: I'm waiting for BE to be merged and then I'll change it to get from `.env.local`, as warned in the slack I'm using development deployed url for now and will change it to get from env after getting backend merged and working

File: kafka-ui-react-app/src/components/App.tsx
Comment (workshur) [inline kafka-ui-react-app/src/components/App.tsx:51]: GlobalSettings

File: kafka-ui-react-app/src/lib/hooks/useTimeFormat.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/hooks/useTimeFormat.ts:10]: (timestamp) =>  
```suggestion
  return (timestamp) =>  formatTimestamp(timestamp, timeStampFormat);
```

File: kafka-ui-react-app/src/lib/hooks/api/timeFormat.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/hooks/api/timeFormat.ts:5]: add default value

**Review by Haarolean — CHANGES_REQUESTED** [review]
1. Date format used currently is a weird one:
image
2. There are no calls to `/api/info/timestampformat` API endpoint as far as I see.

File: kafka-ui-react-app/src/components/Topics/Topic/Statistics/Metrics.tsx
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Statistics/Metrics.tsx:60]: I would suggest to use keep prev version here because we need only 'hh:mm:ss a'

File: kafka-ui-react-app/src/lib/dateTimeHelpers.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/dateTimeHelpers.ts:9]: I don't think we need to transform format from user settings

**Comment by habrahamyanpro** [comment]
is ready

File: kafka-ui-react-app/src/components/Version/Version.tsx
Comment (workshur) [inline kafka-ui-react-app/src/components/Version/Version.tsx:19]:
```suggestion
  const { data: actuatorInfo = {} } = useActuatorInfoStats();
```

File: kafka-ui-react-app/src/components/Version/Version.tsx
Comment (workshur) [inline kafka-ui-react-app/src/components/Version/Version.tsx:45]: Use useQuery with no refetches

File: kafka-ui-react-app/src/components/Version/__tests__/Version.spec.tsx
Comment (workshur) [inline kafka-ui-react-app/src/components/Version/__tests__/Version.spec.tsx:11]: do not use test ids!

File: kafka-ui-react-app/src/components/contexts/GlobalSettingsContext.tsx
Comment (workshur) [inline kafka-ui-react-app/src/components/contexts/GlobalSettingsContext.tsx:4]:
```suggestion
interface GlobalSettingsType {
```

File: kafka-ui-react-app/src/lib/hooks/api/timeFormat.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/hooks/api/timeFormat.ts:12]: add options to reduce refetches

File: kafka-ui-react-app/src/lib/constants.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/constants.ts:52]:
```suggestion
export const BUILD_VERSION_PATTERN = /v\d.\d.\d/;
```

File: kafka-ui-react-app/src/lib/hooks/api/actuatorInfo.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/hooks/api/actuatorInfo.ts:13]: no refetch

File: kafka-ui-react-app/src/lib/hooks/api/brokers.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/hooks/api/brokers.ts:66]: It should be fixed. Pls try to remove

File: kafka-ui-react-app/src/lib/api.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/api.ts:21]:
```suggestion
export const timerStampFormatApiClient = new TimeStampFormatApi(apiClientConf);
```

File: kafka-ui-react-app/src/lib/hooks/api/actuatorInfo.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/hooks/api/actuatorInfo.ts:13]:
```suggestion
  return useQuery(['actuatorInfo'], timerStampFormatApiClient.getTimeStampFormat());
```

File: kafka-ui-react-app/src/lib/hooks/api/timeFormat.ts
Comment (workshur) [inline kafka-ui-react-app/src/lib/hooks/api/timeFormat.ts:12]:
```suggestion
  return useQuery(['settings', 'timestampformat'], timerStampFormatApiClient.getTimeStampFormat());
```

File: kafka-ui-react-app/src/components/Version/Version.tsx
Comment (workshur) [inline kafka-ui-react-app/src/components/Version/Version.tsx:36]: No need to check tag existing again. You have already checked it on L32

**Review by workshur — APPROVED** [review]

File: kafka-ui-react-app/src/components/Version/Version.tsx
Comment (workshur) [reply kafka-ui-react-app/src/components/Version/Version.tsx:36]: Please fix this issue and merge

**Review by workshur — APPROVED** [review]

**Review by workshur — APPROVED** [review]

**Pull request merged**
